lib/Dialect/DC/Transforms/Analysis.cpp

```diff
@@ -0,0 +1,283 @@
+    return DotNode{.nodeType = "fork",
+                   .incoming = valueToName(op.getOperands(), valuesMap, false),
+                   .outgoing = valueToName(op.getResults(), valuesMap, false)};
+  if (auto castOp = llvm::dyn_cast<dc::JoinOp>(op))
+    return DotNode{.nodeType = "join",
+                   .incoming = valueToName(op.getOperands(), valuesMap, false),
+                   .outgoing = valueToName(op.getResults(), valuesMap, false)};
+  if (auto castOp = llvm::dyn_cast<dc::BranchOp>(op))
+    return DotNode{.nodeType = "branch",
+                   .incoming = valueToName(op.getOperands(), valuesMap, false),
+                   .outgoing = valueToName(op.getResults(), valuesMap, false)};
+  if (auto castOp = llvm::dyn_cast<dc::MergeOp>(op))
+    return DotNode{.nodeType = "merge",
+                   .incoming = valueToName(op.getOperands(), valuesMap, false),
+                   .outgoing = valueToName(op.getResults(), valuesMap, false)};
+  if (auto castOp = llvm::dyn_cast<dc::SourceOp>(op))
+    return DotNode{.nodeType = "source",
+                   .incoming = valueToName(op.getOperands(), valuesMap, false),
+                   .outgoing = valueToName(op.getResults(), valuesMap, false)};
+  if (auto castOp = llvm::dyn_cast<dc::SinkOp>(op))
+    return DotNode{.nodeType = "sink",
+                   .incoming = valueToName(op.getOperands(), valuesMap, false),
+                   .outgoing = valueToName(op.getResults(), valuesMap, false)};
+  if (auto castOp = llvm::dyn_cast<dc::SelectOp>(op))
+    return DotNode{.nodeType = "select",
+                   .incoming = valueToName(op.getOperands(), valuesMap, false),
+                   .outgoing = valueToName(op.getResults(), valuesMap, false)};
+  if (auto castOp = llvm::dyn_cast<dc::BufferOp>(op))
+    return DotNode{.nodeType = "buffer",
+                   .incoming = valueToName(op.getOperands(), valuesMap, false),
+                   .outgoing = valueToName(op.getResults(), valuesMap, false)};
+  if (auto castOp = llvm::dyn_cast<dc::FromESIOp>(op))
+    return DotNode{.nodeType = "fromESI",
+                   .incoming = valueToName(op.getOperands(), valuesMap, false),
+                   .outgoing = valueToName(op.getResults(), valuesMap, false)};
+  if (auto castOp = llvm::dyn_cast<dc::PackOp>(op))
+    return DotNode{.nodeType = "pack",
+                   .incoming = valueToName(op.getOperands(), valuesMap, false),
+                   .outgoing = valueToName(op.getResults(), valuesMap, false)};
+  if (auto castOp = llvm::dyn_cast<dc::ToESIOp>(op))
+    return DotNode{.nodeType = "toESI",
+                   .incoming = valueToName(op.getOperands(), valuesMap, false),
+                   .outgoing = valueToName(op.getResults(), valuesMap, false)};
+  if (auto castOp = llvm::dyn_cast<dc::UnpackOp>(op))
+    return DotNode{.nodeType = "unpack",
+                   .incoming = valueToName(op.getOperands(), valuesMap, false),
```

Comment:
Am I correct that the only thing that varies here between the DotNodes that are being returned is nodeType? It might be worth just letting these if statements define what nodeType is, and then feed that to just one call to the DotNode constructor which is probably a lot more concise